### PR TITLE
[ISSUE #11655]fix invalid usage of ProporityQueue in ConfigChangePluginManger

### PR DESCRIPTION
**Pull request by lyf712**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

Fixed an issue  #11655 fix invalid usage of ProporityQueue in ConfigChangePluginMaanger

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: config/src/main/java/com/alibaba/nacos/config/server/aspect/ConfigChangeAspect.java [hunk 1]
```diff
@@ -275,42 +274,41 @@ Object removeConfigAroundRpc(ProceedingJoinPoint pjp, ConfigRemoveRequest reques
...
+        List<ConfigChangePluginService> beforeExecutePluginServices = new ArrayList<>(DEFAULT_BEFORE_LIST_CAPACITY);
+        List<ConfigChangePluginService> afterExecutePluginServices = new ArrayList<>(DEFAULT_AFTER_LIST_CAPACITY);
         
         Object retVal = null;
         Object[] args = pjp.getArgs();
         configChangeRequest.setArg(ConfigChangeConstants.ORIGINAL_ARGS, args);
         
-        for (ConfigChangePluginService ccs : configChangePluginServicePriorityQueue) {
+        for (ConfigChangePluginService ccs : configChangePluginServiceList) {
             if (!isEnabled(ccs)) {
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/aspect/ConfigChangeAspect.java:300]: can  we  move 'isEnable' checking for service to ConfigChangePluginManager？

File: plugin/config/src/test/java/com/alibaba/nacos/plugin/config/ConfigChangePluginManagerTests.java [hunk 2]
```diff
@@ -126,6 +189,19 @@ public void testFindPluginServiceByServiceType() {
...
         Assert.assertFalse(configChangePluginServiceOptional.isPresent());
     }
+
+    private boolean isSorted(List<ConfigChangePluginService> list) {
+        for (int i = 0; i < list.size() - 1; i++) {
+            if (list.get(i).getOrder() > list.get(i + 1).getOrder()) {
+                return false;
+            }
+        }
+        return true;
```
Comment (shiyiyue1102) [inline plugin/config/src/test/java/com/alibaba/nacos/plugin/config/ConfigChangePluginManagerTests.java:205]: can replaced with Lambda

File: plugin/config/src/main/java/com/alibaba/nacos/plugin/config/ConfigChangePluginManager.java [hunk 3]
```diff
@@ -83,8 +84,10 @@ private static void loadConfigChangeServices() {
             // map the relationship of pointcut and plugin service
             addPluginServiceByPointCut(each);
```
Comment (shiyiyue1102) [inline plugin/config/src/main/java/com/alibaba/nacos/plugin/config/ConfigChangePluginManager.java:85]: check  service enabled here ，if not enabled，do not add to the final list , and logs should be printed for what we loaded from service loader and what we ignored if service not enabled

File: config/src/main/java/com/alibaba/nacos/config/server/aspect/ConfigChangeAspect.java [hunk 4]
```diff
@@ -354,19 +352,19 @@ private Object configChangeServiceHandle(ProceedingJoinPoint pjp,
         return retVal;
     }
     
-    private PriorityQueue<ConfigChangePluginService> getPluginServicePriorityQueue(
+    private List<ConfigChangePluginService> getPluginServices(
             ConfigChangePointCutTypes configChangePointCutType) {
-        PriorityQueue<ConfigChangePluginService> pluginServicePriorityQueue = ConfigChangePluginManager
-                .findPluginServiceQueueByPointcut(configChangePointCutType);
+        List<ConfigChangePluginService> pluginServicePriorityQueue = ConfigChangePluginManager
```
Comment (shiyiyue1102) [inline config/src/main/java/com/alibaba/nacos/config/server/aspect/ConfigChangeAspect.java:357]: pluginServicePriorityQueue should  be renamed to pluginServicePriorityList

**Comment by shiyiyue1102** [comment]
> ## [Codecov](https://app.codecov.io/gh/alibaba/nacos/pull/11668?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=alibaba) Report
> Attention: `12 lines` in your changes are missing coverage. Please review.
> 
> > Comparison is base [(`a7d1c35`)](https://app.codecov.io/gh/alibaba/nacos/commit/a7d1c35f85362a4538e81a721ab94b6bceeb714e?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=alibaba) 66.75% compared to head [(`a69f5fe`)](https://app.codecov.io/gh/alibaba/nacos/pull/11668?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=alibaba) 66.71%.
> 
> Additional details and impacted files

pay attention to coverage decline

File: plugin/config/src/main/java/com/alibaba/nacos/plugin/config/ConfigChangePluginManager.java (hunk 3 above)
Comment (lyf712) [reply plugin/config/src/main/java/com/alibaba/nacos/plugin/config/ConfigChangePluginManager.java:85]: If you load here, it won't be able to fetch dynamically updated config like AuthConfigs?

Comment (shiyiyue1102) [reply plugin/config/src/main/java/com/alibaba/nacos/plugin/config/ConfigChangePluginManager.java:85]: > If you load here, it won't be able to fetch dynamically updated config like AuthConfigs?
ok, you are right, just ignore this comment

**Review by shiyiyue1102 — APPROVED** [review]

**Pull request merged**